lms/djangoapps/course_home_api/dates/v1/views.py

```diff
@@ -83,14 +84,18 @@ def get(self, request, *args, **kwargs):
         monitoring_utils.set_custom_attribute('is_staff', request.user.is_staff)
 
         course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        is_staff = bool(has_access(request.user, 'staff', course_key))
 
         _, request.user = setup_masquerade(
             request,
             course_key,
-            staff_access=has_access(request.user, 'staff', course_key),
+            staff_access=is_staff,
             reset_masquerade_data=True,
         )
 
+        if not CourseEnrollment.is_enrolled(request.user, course_key) and not is_staff:
```

Comment:
Nit: Switch the order here so if `not is_staff` is False, it will short circuit. Will save a potential query to look up the enrollment in those cases.